### PR TITLE
DDD-22 Add Swagger

### PR DESCRIPTION
**Pull request by Zygis0321**
added swagger

### CODE REVIEW

**Review by KornelisMax — APPROVED** [review]
Great job 🚀 Just wanted to give few insights for pull request naming and commit message structure:

- PR name should briefly describe content and it should be written in imperative present tense (for example: "Add swagger", "Create swagger documentation" etc.). Avoid random "unique identifiers" such as "attempt2";
- Use commit messages (`git commit -m`) to briefly describe your change (use body to explain something in more detail, but usually commit message is enough to summarize the change);
- Commit message similarly to PR name, should be written in imperative present tense (for example: `git commit -m "DDD-22 Add swagger dependencies"`);

We're creating "dummy" project, but it is quite important do develop good habits along the way too 💪

File: src/main/java/lt/vu/web/api/controller/packageSize/ListPackageSizeController.java [hunk 1]
```diff
@@ -16,12 +19,15 @@
 
 @Path("/package-size")
 @RequestScoped
+@Api(value= "List package size ") // swagger minimum (controller name)
```
Comment (giplonis) [inline src/main/java/lt/vu/web/api/controller/packageSize/ListPackageSizeController.java]: Blank space at the end of line and we don't need these redundant comments

File: src/main/java/lt/vu/web/api/controller/packageSize/ListPackageSizeController.java [hunk 2]
```diff
@@ -16,12 +19,15 @@
...
 public class ListPackageSizeController extends AbstractApiController<List<PackageSizeDTO>> {
 
     @Inject
     private PackageSizeRepository packageSizeRepository;
 
     @GET
+    @ApiOperation(value = "Retrieves valid package sizes",         //swagger optional (endpoint summary)
+                  notes = "Return some json to the client")  //swagger optional (description)
```
Comment (giplonis) [inline src/main/java/lt/vu/web/api/controller/packageSize/ListPackageSizeController.java]: No need for these redundant notes. I also tried adding `response` so it displays the model structure (which is the main point of having swagger) and it crashes.

**Review by giplonis — CHANGES_REQUESTED** [review]
I could not get this to work (after adding `response` models). Why use such an unpopular lib while swagger offers [official package](https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Getting-started#quick-start)?

I suggest we go with official dependencies:
```
<dependency>
       <groupId>io.swagger.core.v3</groupId>
       <artifactId>swagger-jaxrs2</artifactId>
       <version>2.1.1</version>
</dependency>
<dependency>
       <groupId>io.swagger.core.v3</groupId>
       <artifactId>swagger-jaxrs2-servlet-initializer</artifactId>
       <version>2.1.1</version>
</dependency>
```

This way at least we know it's maintained and up to date. However, I'm getting same error while trying to describe api models:
image

**Comment by giplonis** [comment]
After some investigation I managed to make it work:
- Installed official Swagger packages
- Created an endpoint to load Swagger documentation static files ([swagger-ui](https://github.com/swagger-api/swagger-ui))
- Updated annotations with relevant info ([official documentation](https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Getting-started))

image

Documentation is available at http://localhost:8080/delivery-service/api/v1/api-doc

**Review by giplonis — APPROVED** [review]

**Pull request merged**